### PR TITLE
quantify sphericity functions

### PR DESCRIPTION
**Pull request by mumuMia**
This commit added BiomedImage.py, test_biomedimage.py, and LabelConnectivity files. Note, I have tried to use the LabelConnectivity object but it kept giving me the object of types set is not JSON serializable. So in this commit, I'm still using the string input for labelConnectivity. Thank you!

### CODE REVIEW

File: cvpy/tests/test_biomedimage.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        # Compute the sphericity
+        biomed.quantify_sphericity(image_table = input, 
+                                   use_spacing = True, 
+                                   input_background = 0, 
+                                   label_connectivity = 'FACE', 
+                                #    label_connectivity = LabelConnectivity.FACE, 
+                                   sphericity= output_sph)
+
+        self.assertTrue(output_sph is not None)
```
Comment (ptelang) [inline cvpy/tests/test_biomedimage.py:59]: I think we should verify if the output table contains expected sphericity value. We can query the sphericity value using fedsql query and see if it equal to expected value.

File: cvpy/tests/test_biomedimage.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import os
+import unittest
+import xmlrunner
+import swat
+import sys
+import numpy as np
+from cvpy.biomedimage.BiomedImage import BiomedImage
+# from cvpy.biomedimage.LabelConnectivity import LabelConnectivity
+
+class TestBiomedimage(unittest.TestCase):
```
Comment (ptelang) [inline cvpy/tests/test_biomedimage.py:28]: TestBiomedimage => TestBiomedImage

File: cvpy/biomedimage/BiomedImage.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import numpy as np
+from typing import List
+from warnings import warn
+from swat import CAS, CASTable
+
+# from cvpy.biomedimage.LabelConnectivity import LabelConnectivity
+
+class BiomedImage(object):
+    '''
+    This is the biomedimage class, with several biomedimage-related functions implemented under it.
```
Comment (ptelang) [inline cvpy/biomedimage/BiomedImage.py:32]: This class implements biomedical image processing functions.

File: cvpy/biomedimage/BiomedImage.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+# from cvpy.biomedimage.LabelConnectivity import LabelConnectivity
+
+class BiomedImage(object):
+    '''
+    This is the biomedimage class, with several biomedimage-related functions implemented under it. 
+    '''
+
+    def __init__(self, cas_session: CAS = None) -> None:
+        '''
+        Constructor for project class
```
Comment (ptelang) [inline cvpy/biomedimage/BiomedImage.py:37]: Constructor for project class => Constructor for BiomedImage class

File: cvpy/biomedimage/BiomedImage.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    def quantify_sphericity(self, image_table: CASTable, use_spacing: bool, input_background: float,
+                            label_connectivity: str, sphericity: CASTable) ->None:
+        ''''
+        Quantify the sphericity for the given component from a CAS table. 
+        Parameters:
+        ----------
+        image_table: CAS table
+             CAS table image includes the image binaries 
+        use_spacing: Bool
```
Comment (ptelang) [inline cvpy/biomedimage/BiomedImage.py:59]: use_spacing: Bool => use_spacing: bool

File: cvpy/biomedimage/BiomedImage.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+             Specifies whether use spacing for the sphericity
+        input_background: float
+             Specifies the background value in input images.
+        label_connectivity:
```
Comment (ptelang) [inline cvpy/biomedimage/BiomedImage.py:63]: label_connectivity: LabelConnectivity

File: cvpy/biomedimage/BiomedImage.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+                                              inputbackground=input_background,
+                                              casout=dict(name='quantify', replace=True),
+                                              )
+
+        ## Compute sphericity based on perimeter and volume of the lesion
+        conn.fedsql.execdirect(
+            "create table sphericity {option replace=true} as "
+            "select _path_,_perimeter_,_content_, (power(pi(), 1.0/3.0) * power(6*_content_, 2.0/3.0))/_perimeter_ as "
+            "sphericity from quantify"
+            )
```
Comment (ptelang) [inline cvpy/biomedimage/BiomedImage.py:108]: I am assuming "quantify" is a temporary intermediate table created by this function. If that is correct, it should be deleted here.

File: cvpy/biomedimage/BiomedImage.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        conn = self._cas_session
+       
+        ## Load the actionsets
+        conn.loadactionset('image')
+        conn.loadactionset('biomedimage')
+        conn.loadactionset('fedsql')
+        
+        ## Quantify the volume and perimeter of the given component.
+        conn.biomedimage.quantifybiomedimages(images= dict(table= image_table),
+                                              copyvars=["_path_"],
```
Comment (ptelang) [inline cvpy/biomedimage/BiomedImage.py:93]: I will suggest using single quotes for strings throughout. Use double quotes only if the string already contains a single quote. 

"_path_" => '_path_'

**Review by ptelang — COMMENTED** [review]
Hi Mu, Great job with this implementation. I have several mostly minor comments. Thanks!

File: cvpy/biomedimage/BiomedImage.py [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    @property
+    def cas_session(self) -> CAS:
+        return self._cas_session
+
+    @cas_session.setter
+    def cas_session(self, cas_session) -> None:
+        self._cas_session = cas_session
+
+
+    def quantify_sphericity(self, image_table: CASTable, use_spacing: bool, input_background: float,
```
Comment (fijoy) [inline cvpy/biomedimage/BiomedImage.py:51]: We talked about creating an input table object that different functions will have as an argument (as opposed to directly using the SWAT CASTable object). Did we decide to do that later on? I can't remember. If we're doing that later on, can you go ahead and create a story to do that and put it in a future sprint?

**Review by fijoy — COMMENTED** [review]
Thanks Mu for working on this. I have 1 comment.

File: cvpy/biomedimage/BiomedImage.py (hunk 9 above)
Comment (ptelang) [reply cvpy/biomedimage/BiomedImage.py:51]: @fijoy, @mumuMia - I created an ImageTable object in my PR. 

https://github.com/sassoftware/python-cvpy/pull/14

**Comment by mumuMia** [comment]
Hi All. I have just submitted new changes for biomedimage.py and its tests. Thank you for taking the time to review them.

**Pull request closed without merging**